site/components/pages/Pledge/types/index.ts

```diff
@@ -0,0 +1,22 @@
+import * as yup from "yup";
+
+import { formSchema } from "../lib";
+
+export type { putPledgeParams } from "../lib/putPledge";
+
+export type Footprint = {
+  timestamp: number;
+  total: number;
+};
+
+export type Pledge = {
+  id: string;
+  ownerAddress: string;
+  name: string;
+  nonce: string;
+  description: string;
+  methodology: string;
+  footprint: Footprint[];
+};
+
+export type PledgeFormValues = yup.InferType<typeof formSchema>;
```

Comment:
hm related to my other comment what's the difference between PledgeFormValues and Pledge?

Reply:
ah, footprint is an array on the server and string in the form? I think I get it. Ignore my other comment about formSchema